XIPs/zip-15-attachment-content-type.md

```diff
@@ -0,0 +1,60 @@
+---
+xip: 15
+title: Attachment Content Type
+description: Content type for basic attachments
+author: Pat Nakajima
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-02-15
+---
+
+## Abstract
+
+This XRC proposes a new content type for messages that do not consist of text. It attempts to include only the bare minimum for clients to determine how to display such messages in order to provide flexibility for more rich types in the future.
```

Comment:
Given the results of [our most recent benchmarks](https://github.com/xmtp/xmtp-js/pull/273), I wonder how hard we should be pushing client developers to attach large-ish attachments to messages.

Reply:
If the client wants to store it on our network, it has to be a message. I could imagine a world where we generate one-off encryption keys and store those in a message, and then use those keys to decrypt content stored elsewhere. That would be E2EE but would allow attachments of unlimited size.

Of course, then we have to figure out where and how to host the attachment.

Reply:
I think that could be a rad new content type as well, something like a remote attachment. I still think there's probably room for smaller attachments that are inline. Sort of like how iMessage lets you send pics and videos up to a certain size, then suggests uploading to iCloud for larger stuff.

Reply:
Wrote a proposal for a remote attachment type here: https://github.com/xmtp/XIPs/pull/18.

Reply:
Just wondering, if I send a 1MB image on the network, it will be part of the encoded `message` field on the `envelope.message` field right?
Since we send the envelope via notification to the mobile apps, should we take into account the size limit of notification payloads?